test/functional/wallet_migration.py

```diff
@@ -1196,6 +1196,60 @@ def test_disallowed_p2wsh(self):
 
         wallet.unloadwallet()
 
+    def test_miniscript(self):
+        # It turns out that due to how signing logic works, legacy wallets that have valid miniscript witnessScripts
+        # and the private keys for them can still sign and spend them, even though output scripts involving them
+        # as a witnessScript would not be detected as ISMINE_SPENDABLE.
```

Comment:
e41f47597d86022d5a9947ee0b08151d4336cffc: Is this only intended for users who called `importmulti` on a legacy wallet? Or could those scripts have ended up in legacy wallets in other ways?

If the latter is the case, do you mean that only modern node versions with miniscript support can sign them? Or could pre-miniscript nodes spend them as well? Might be worth illustrating with an older node (< v25?), if the `miniscript ` wallet here can be loaded by that older version.

Reply:
I believe the only way is through `importmulti`.

Versions without miniscript cannot sign for miniscript. I feel like that's obvious and we don't need a test for that.

Reply:
That's not going to obvious for most people looking at this in the future, so I think it's worth a comment (a test might be overkill).